[UXE-7251] method route validator guide (#1666)
aziontech/docs: src/content/docs/en/pages/guides/marketplace/integrations/method-route-validator.mdx

---
title: Method and Route Validator
description: >-
  This function can be used to protect an application by blocking requests based on the URI and method used.
meta_tags: API, website, security, URI, method, route, validator
namespace: docs_use_case_method_and_route_validator
permalink: /documentation/products/guides/method-and-route-validator/
---

import Tag from 'primevue/tag';

<Tag severity="info" client:only="vue">
Preview
</Tag>

The **Method and Route Validator** integration can be used to protect your application by blocking requests based on the URI and method used. It works by validating the arguments received by the function and checking if the request matches the patterns defined in any of the values.

> With this integration, you can define a list of routes and which method each route expects. Each time the function is executed, it will follow the following steps:

Reviewer: follow the following...
```suggestion
With this integration, you can define a list of routes and which method each route expects. Each time the function is executed, it'll perform the following steps:
```

1. Validate the arguments passed to the function.
2. Check if the request matches the patterns defined for the routes.
  - If there is a match, the function will perform a user-defined blocking action.
  - If there is no match, the function can be configured to continue the request or block it.

> 3. If the arguments are invalid, it will write a log message and continue the request.

Reviewer: ```suggestion
3. If the arguments are invalid, it'll write a log message and continue the request.
```

| `equals` | The path must be equal to the one defined in the `path` argument. |
| `contains` | The path must contain the one defined in the `path` argument. |
| `regex` | The path must match the regular expression defined in the `path` argument. |

:::note
The `routes.methods` argument expects values in uppercase. Example: `["GET", "POST"]`.
:::

6. Click the **Save** button.

### Setting up the Rules Engine

To finish, you have to set up a rule in **Rules Engine** to configure the **criteria** and the **behavior** to run the function.

Still in the **Edge Firewall** page:

1. Select the **Rules Engine** tab.
2. Click the **+ Rule Engine** button.
3. Give a name to the rule.
4. Select a *criteria* to run and catch the domains that you want to run the integration on.
  - Example: if `Host` *matches* `yourdomain.com`.
5. Below, select a *behavior* to the *criteria*. In this case, it'll be **Run Function**. 
    - Select the adequate function according to the name you gave it during the instantiation step.
6. Click the **Save** button.

On the Console, you must now configure your domain so your edge firewall protects it.

7. On the **Products menu**, select **Domains**.

> 8. Click on the domain you want to protect with your Method and Route Validator function.

Reviewer: ```suggestion
8. Click on the domain you want to protect with your **Method and Route Validator** function.
```
